drivers/cisco/ise/guests.cr

```diff
@@ -0,0 +1,152 @@
+require "xml"
+
+# Tested with Cisco ISE API v2.2
+# https://developer.cisco.com/docs/identity-services-engine/3.0/#!guest-user/resource-definition
+# However, should work and conform to v1.4 requirements
+# https://www.cisco.com/c/en/us/td/docs/security/ise/1-4/api_ref_guide/api_ref_book/ise_api_ref_guest.html#79039
+
+class Cisco::Ise::Guests < PlaceOS::Driver
+  # Discovery Information
+  descriptive_name "Cisco ISE Guest Control"
+  generic_name :Guests
+  uri_base "https://ise-pan:9060/ers/config"
+
+  default_settings({
+    username:   "user",
+    password:   "pass",
+    portal_id:  "Required, ask cisco ISE admins",
+    timezone:   "Australia/Sydney",
+    guest_type: "Required, ask cisco ISE admins for valid subset of values",                              # e.g. Contractor
+    location:   "Required for ISE v2.2, ask cisco ISE admins for valid value. Else, remove for ISE v1.4", # e.g. New York
+    custom_data: {} of String => JSON::Any::Type
+  })
+
+  @basic_auth : String = ""
+  @portal_id : String = ""
+  @sms_service_provider : String? = nil
+  @guest_type : String = "default_guest_type"
+  @timezone : Time::Location = Time::Location.load("Australia/Sydney")
+  @location : String? = nil
+  @custom_data = {} of String => JSON::Any::Type
+
+  TYPE_HEADER = "application/vnd.com.cisco.ise.identity.guestuser.2.0+xml"
+  TIME_FORMAT = "%m/%d/%Y %H:%M"
+
+  def on_load
+    on_update
+  end
+
+  def on_update
+    @basic_auth = "Basic #{Base64.strict_encode("#{setting?(String, :username)}:#{setting?(String, :password)}")}"
+    @portal_id = setting?(String, :portal_id) || "portal101"
+    @guest_type = setting?(String, :guest_type) || "default_guest_type"
+    @location = setting?(String, :location)
+    @sms_service_provider = setting?(String, :sms_service_provider)
+
+    time_zone = setting?(String, :timezone).presence
+    @timezone = Time::Location.load(time_zone) if time_zone
+    @custom_data = setting?(Hash(String, JSON::Any::Type), :custom_data) || {} of String => JSON::Any::Type
+  end
+
+  def create_guest(
+    event_start : Int64,
+    attendee_email : String,
+    attendee_name : String,
+    company_name : String? = nil,         # Mandatory but driver will extract from email if not passed
+    phone_number : String = "0123456789", # Mandatory, use a fake value as default
+    sms_service_provider : String? = nil, # Use this param to override the setting
+    guest_type : String? = nil,           # Mandatory but use this param to override the setting
+    portal_id : String? = nil             # Mandatory but use this param to override the setting
+  )
+    # Determine the name of the attendee for ISE
+    guest_names = attendee_name.split
+    first_name_index_end = guest_names.size > 1 ? -2 : -1
+    first_name = guest_names[0..first_name_index_end].join(' ')
+    last_name = guest_names[-1]
```

Comment:
took me a minute to grok this but yeah that will work.
FYI I think when the input name is a string with no spaces then last_name would be the same as first_name instead of being null - that may be fine for your use case as long as you're aware that it could happen

Reply:
Sorry it also confused me a bit looking over at it again especially because of the negative indexes with first_name_index_end but I think it is the cleanest way to do this. I probably should of added a comment.

Reply:
I think I accepted that last_name would be the same as first_name in a case where the `attendee_name = "Phil"` since that would be pretty rare I think